rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -35,13 +36,20 @@ object BoolNormalizeMatcher {
 }
 
 object GroundNormalizeMatcher {
-  def normalizeMatch(g: AbsynGround): Expr =
+  def normalizeMatch[M[_]](g: AbsynGround)(implicit M: MonadError[M, Throwable]): M[Expr] =
     g match {
-      case gb: GroundBool => BoolNormalizeMatcher.normalizeMatch(gb.boolliteral_)
+      case gb: GroundBool => Expr(BoolNormalizeMatcher.normalizeMatch(gb.boolliteral_)).pure[M]
       case gi: GroundInt =>
-        GInt(gi.longliteral_.toLong) //TODO raise NumberFormatException in a pure way
-      case gs: GroundString => GString(stripString(gs.stringliteral_))
-      case gu: GroundUri    => GUri(stripUri(gu.uriliteral_))
+        M.fromTry(
+            Try(gi.longliteral_.toLong).adaptError {
+              case e: NumberFormatException => NormalizerError(e.getMessage)
+            }
+          )
+          .map { long =>
+            Expr(GInt(long))
+          }
+      case gs: GroundString => Expr(GString(stripString(gs.stringliteral_))).pure[M]
+      case gu: GroundUri    => Expr(GUri(stripUri(gu.uriliteral_))).pure[M]
```

Comment:
A thought: @marcin-rzeznicki's combo of:
```
1. remove `import cats.implicits._`
2. add `import cats.syntax.all._`
3. add the needed `import cats.instances.$datatype._`
```
can usually be done in less than 5 mins per file. Maybe it makes sense to do it the moment we see a type annotation like those in `.pure[M]`?

Reply:
I've taken a few minutes to try this in the normalizer. It's not yielding the same reductions as this method has in other parts of the code base. I'm not convinced it merits investigating why right now.